src/file_scanner/arrow_multi_file_info.cpp

```diff
@@ -0,0 +1,250 @@
+    lstate.init_input = make_uniq<TableFunctionInitInput>(
+        *lstate.local_arrow_function_data, reader->column_indexes,
+        gstate.global_state.projection_ids, reader->filters);
+  } else {
+    lstate.init_input = make_uniq<TableFunctionInitInput>(
+        *lstate.local_arrow_function_data, gstate.global_state.column_indexes,
+        gstate.global_state.projection_ids, reader->filters);
+  }
+  lstate.local_arrow_global_state =
+      ArrowTableFunction::ArrowScanInitGlobal(context, *lstate.init_input);
+  lstate.local_arrow_local_state =
+      ArrowTableFunction::ArrowScanInitLocal(lstate.execution_context, *lstate.init_input,
+                                             lstate.local_arrow_global_state.get());
+  lstate.table_function_input = make_uniq<TableFunctionInput>(
+      lstate.local_arrow_function_data.get(), lstate.local_arrow_local_state.get(),
+      lstate.local_arrow_global_state.get());
+  return true;
+}
+
+void ArrowMultiFileInfo::Scan(ClientContext& context, BaseFileReader& reader,
+                              GlobalTableFunctionState& global_state,
+                              LocalTableFunctionState& local_state, DataChunk& chunk) {
+  auto& lstate = local_state.Cast<ArrowFileLocalState>();
+  ArrowTableFunction::ArrowScanFunction(context, *lstate.table_function_input, chunk);
+}
+
+void ArrowMultiFileInfo::FinishFile(ClientContext& context,
+                                    GlobalTableFunctionState& global_state,
+                                    BaseFileReader& reader) {}
+
+void ArrowMultiFileInfo::FinishReading(ClientContext& context,
+                                       GlobalTableFunctionState& global_state,
+                                       LocalTableFunctionState& local_state) {}
+
+unique_ptr<NodeStatistics> ArrowMultiFileInfo::GetCardinality(
+    const MultiFileBindData& bind_data, idx_t file_count) {
+  return make_uniq<NodeStatistics>();
+}
+
+unique_ptr<BaseStatistics> ArrowMultiFileInfo::GetStatistics(ClientContext& context,
+                                                             BaseFileReader& reader,
+                                                             const string& name) {
+  throw InternalException("Unimplemented ArrowMultiFileInfo GetStatistics method");
+}
+
+double ArrowMultiFileInfo::GetProgressInFile(ClientContext& context,
+                                             const BaseFileReader& reader) {
+  auto& file_scan = reader.Cast<ArrowFileScan>();
+  if (!file_scan.factory->reader) {
+    // We are done with this file
+    return 100;
+  }
+  auto file_reader =
+      reinterpret_cast<IPCFileStreamReader*>(file_scan.factory->reader.get());
+  return file_reader->GetProgress();
+}
+
+void ArrowMultiFileInfo::GetVirtualColumns(ClientContext&, MultiFileBindData&,
+                                           virtual_column_map_t& result) {
+  result.insert(
+      make_pair(COLUMN_IDENTIFIER_EMPTY, TableColumn("", LogicalType::BOOLEAN)));
```

Comment:
FYI this is where the `COLUMN_IDENTIFIER_EMPTY` comes from, it should not be emitted unless this is specifically declared

Reply:
Aha, so I dont need the other PR then?